### ISSUE TITLE
Exception on reading MemoryUsage (possible wrong max memory value)

### ISSUE THREAD

**Issue by joggeli34** [oracle/graal#6814]
**Describe the issue**
With the newest graalvm release, it is now possible to read out the memory usage. We use quarkus with micrometer, where we get following exception:

```
2023-06-15 14:18:58,409 WARN  [io.mic.cor.ins.int.DefaultGauge] (executor-thread-3) Failed to apply the value function for the gauge 'jvm.memory.used'. Note that subsequent logs will be logged at debug level.: java.lang.IllegalArgumentException: committed = 8573528 should be < max = 8388608
	at java.management@17.0.7/java.lang.management.MemoryUsage.<init>(MemoryUsage.java:166)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.genscavenge.AbstractMemoryPoolMXBean.memoryUsage(AbstractMemoryPoolMXBean.java:75)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.genscavenge.GenScavengeMemoryPoolMXBeans$EdenMemoryPoolMXBean.getUsage(GenScavengeMemoryPoolMXBeans.java:94)
	at io.micrometer.core.instrument.binder.jvm.JvmMemory.getUsage(JvmMemory.java:76)
	at io.micrometer.core.instrument.binder.jvm.JvmMemory.getUsageValue(JvmMemory.java:66)
	at io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics.lambda$bindTo$0(JvmMemoryMetrics.java:82)
	at io.micrometer.core.instrument.internal.DefaultGauge.value(DefaultGauge.java:53)
	at io.micrometer.prometheus.PrometheusMeterRegistry.lambda$newGauge$5(PrometheusMeterRegistry.java:331)
	at io.micrometer.prometheus.MicrometerCollector.collect(MicrometerCollector.java:75)
	at io.prometheus.client.Collector.collect(Collector.java:45)
	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.findNextElement(CollectorRegistry.java:204)
	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:219)
	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:152)
	at io.prometheus.client.exporter.common.TextFormat.write004(TextFormat.java:71)
	at io.prometheus.client.exporter.common.TextFormat.writeFormat(TextFormat.java:53)
	at io.micrometer.prometheus.PrometheusMeterRegistry.scrape(PrometheusMeterRegistry.java:154)
	at io.micrometer.prometheus.PrometheusMeterRegistry.scrape(PrometheusMeterRegistry.java:149)
	at io.micrometer.prometheus.PrometheusMeterRegistry.scrape(PrometheusMeterRegistry.java:120)
	at io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler.doHandle(PrometheusHandler.java:63)
	at io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler.handle(PrometheusHandler.java:43)
	at io.quarkus.micrometer.runtime.export.handlers.PrometheusHandler.handle(PrometheusHandler.java:18)
	at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
	at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@17.0.7/java.lang.Thread.run(Thread.java:833)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:807)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:210)
```

We used as run parameters "-Xmx30m -Xmn10m". So there is also a difference between the 10m and the 8m reported here.

**Steps to reproduce the issue**
We don't have a reproducer yet separate from our production code.

**Describe GraalVM and your environment:**
 - GraalVM version **CE 17.0**
 - JDK major version: **17**
 - OS: **Linux**
 - Architecture: **AMD64**
 - Quarkus: **3.1.1**

**More details**
Additionally, we saw that the eden-space max value is not always readed out as the same, but we are not sure if this also could be some aggregation issue (with setting -Xmx100m):
![image](https://github.com/oracle/graal/assets/34470368/88fd7ecc-3f4f-42f4-80ef-ece6f1fff7e5)

**Comment by joggeli34** [comment]
I haven't had the time to create a reproducer. I will try to find time next week.

**Comment by 0xabadea** [comment]
Attaching a Spring Boot project that consistently reproduces the issue for me: [issue6814.zip](https://github.com/oracle/graal/files/12172341/issue6814.zip) To reproduce:

1. Unzip the attached zip file.
2. Execute `./gradlew bootBuildImage`
3. In a separate terminal execute `while true; do curl localhost:9080/actuator/prometheus; done`.
4. Execute `docker run -m 256m -p 9080:9080 docker.io/library/issue6814:0.0.1-SNAPSHOT -Xmx128m -XX:MaxDirectMemorySize=40m`

For me, the application starts up and immediately throws an exception very similar to the one in the original description:

```
2023-07-26T12:16:55.796Z  WARN [issue6814,10fcbf35e2e21545adde41bf5c61362a,44433b2bc18acabc] 1 --- [oundedElastic-1] i.m.c.instrument.internal.DefaultGauge   : Failed to apply the value function for the gauge 'jvm.memory.committed'. Note that subsequent logs will be logged at debug level.

java.lang.IllegalArgumentException: committed = 27262976 should be < max = 14680064
	at java.management@17.0.7/java.lang.management.MemoryUsage.<init>(MemoryUsage.java:166) ~[na:na]
	at com.oracle.svm.core.genscavenge.AbstractMemoryPoolMXBean.memoryUsage(AbstractMemoryPoolMXBean.java:75) ~[com.example.issue6814.Issue6814ApplicationKt:na]
	at com.oracle.svm.core.genscavenge.GenScavengeMemoryPoolMXBeans$EdenMemoryPoolMXBean.getUsage(GenScavengeMemoryPoolMXBeans.java:94) ~[na:na]
	at io.micrometer.core.instrument.binder.jvm.JvmMemory.getUsage(JvmMemory.java:76) ~[na:na]
```

Please note the exception is not identical: the original description refers to the `jvm.memory.used` gauge, while the sample project refers to `jvm.memory.committed`. However, both exceptions happen when calling `GenScavengeMemoryPoolMXBeans$EdenMemoryPoolMXBean.getUsage`, so hopefully the sample project is relevant.

**Comment by joggeli34** [comment]
Any news about this? It still happens on graalvm jdk-21

**Comment by oubidar-Abderrahim** [comment]
Should be fixed by #6930
